lib/compat/wordpress-5.9/class-gutenberg-rest-global-styles-controller.php

```diff
@@ -218,18 +218,46 @@ protected function prepare_item_for_database( $request ) {
 	public function prepare_item_for_response( $post, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
 		$config                           = json_decode( $post->post_content, true );
 		$is_global_styles_user_theme_json = isset( $config['isGlobalStylesUserThemeJSON'] ) && true === $config['isGlobalStylesUserThemeJSON'];
-		$result                           = array(
-			'id'       => $post->ID,
-			'settings' => ! empty( $config['settings'] ) && $is_global_styles_user_theme_json ? $config['settings'] : new stdClass(),
-			'styles'   => ! empty( $config['styles'] ) && $is_global_styles_user_theme_json ? $config['styles'] : new stdClass(),
-			'title'    => array(
-				'raw'      => $post->post_title,
-				'rendered' => get_the_title( $post ),
-			),
-		);
-		$result                           = $this->add_additional_fields_to_object( $result, $request );
-		$response                         = rest_ensure_response( $result );
-		$links                            = $this->prepare_links( $post->id );
+
+		$fields = $this->get_fields_for_response( $request );
+
+		// Base fields for every post.
+		$data = array();
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( rest_is_field_included( 'title', $fields ) ) {
+			$data['title'] = array();
+		}
+		if ( rest_is_field_included( 'title.raw', $fields ) ) {
+			$data['title']['raw'] = $post->post_title;
+		}
+		if ( rest_is_field_included( 'title.rendered', $fields ) ) {
+			add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
```

Comment:
protected style variations is not something that is meaningful for the global styles IMO, so I'd rather remove all these "protected" handling.

Reply:
This is how core handles protected titles. We can't do much about it.

Reply:
Global styles posts won't be password protected, so I'm not sure we need to handle it.

Reply:
Better safe than sorry?